src/main/java/seedu/loyaltylift/model/order/StatusUpdate.java

```diff
@@ -53,6 +53,19 @@ public StatusUpdate nextStatusUpdate() {
         return nextStatusUpdate(LocalDate.now());
     }
 
+    /**
+     * Returns a new {@code StatusUpdate} which StatusValue is "cancelled"
+     * with the given date.
+     * @param date A valid date.
+     * @return A new StatusUpdate.
+     */
+    public StatusUpdate cancelOrder(LocalDate date) {
```

Comment:
Does not feel like this method is necessary. This abstraction is not used anywhere other than `Status` and the guard statements are duplicated. Creating a `StatusUpdate` with a cancelled `StatusValue` directly in `Status` feels sufficient to me.